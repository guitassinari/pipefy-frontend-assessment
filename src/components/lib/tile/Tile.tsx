import React from 'react'
import { TileBackground } from './Tile.style'

interface TileProps {
  bgColor?: string
}

const Tile: React.FC<TileProps & React.HTMLAttributes<HTMLDivElement>> = ({ children, ...rest }) => (
  <TileBackground {...rest}>
    <div>
      {children}
    </div>
  </TileBackground>
)

export default Tile