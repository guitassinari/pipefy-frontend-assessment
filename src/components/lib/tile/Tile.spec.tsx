import React from 'react';
import { render } from '@testing-library/react'
import Tile from '.';

describe('PipeTile', () => {
  it('matches the snapshot', () => {
    const component = render(<Tile>Teste</Tile>)

    expect(component.asFragment()).toMatchSnapshot()
  })

  it('shows the received children', () => {
    const component = render(<Tile>Teste</Tile>)

    expect(component.getByText('Teste')).toBeInTheDocument()
  })

  it('allows custom styling', () => {
    const component = render(<Tile className="className">Teste</Tile>)

    expect(component.asFragment()).toMatchSnapshot()
  })

  it('allows setting the background color', () => {
    const component = render(<Tile bgColor="blue">Teste</Tile>)

    expect(component.asFragment()).toMatchSnapshot()
  })
})
