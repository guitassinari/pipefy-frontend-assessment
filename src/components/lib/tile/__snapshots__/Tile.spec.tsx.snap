// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PipeTile allows custom styling 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL kgCPtU className"
  >
    <div>
      Teste
    </div>
  </div>
</DocumentFragment>
`;

exports[`PipeTile allows setting the background color 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL kJwlWd"
  >
    <div>
      Teste
    </div>
  </div>
</DocumentFragment>
`;

exports[`PipeTile matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL kgCPtU"
  >
    <div>
      Teste
    </div>
  </div>
</DocumentFragment>
`;
